boa/src/bytecompiler.rs

```diff
@@ -858,24 +980,31 @@ impl ByteCompiler {
                 for decl in list.as_ref() {
                     match decl {
                         Declaration::Identifier { ident, .. } => {
+                            if ident.as_ref() == "arguments" {
+                                self.code_block.lexical_name_argument = true;
+                            }
+
                             let index = self.get_or_insert_name(ident.as_ref());
-                            self.emit(Opcode::DefConst, &[index]);
 
                             if let Some(expr) = decl.init() {
                                 self.compile_expr(expr, true);
-                                self.emit(Opcode::InitLexical, &[index]);
-                            };
+                                self.emit(Opcode::DefInitConst, &[index]);
+                            } else {
+                                self.emit(Opcode::DefConst, &[index]);
+                            }
```

Comment:
I think this `else` is unreachable and we might not need the `DefConst` opcode, correct me if I'm wrong, but I think `const` declarations require an initializer.

Reply:
Const declarations require an initializer always, yes. It's checked at parsing time.

Reply:
I changed it to expect the initializer and removed to unused opcode. Nice catch.